app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/settings/accountsecurity/deleteaccount/DeleteAccountViewModel.kt

```diff
@@ -74,46 +85,63 @@ class DeleteAccountViewModel @Inject constructor(
     private fun handleDeleteAccountConfirmDialogClick(
         action: DeleteAccountAction.DeleteAccountConfirmDialogClick,
     ) {
-        mutableStateFlow.update {
-            it.copy(dialog = DeleteAccountState.DeleteAccountDialog.Loading)
-        }
         viewModelScope.launch {
-            val result = authRepository.deleteAccountWithMasterPassword(action.masterPassword)
-            sendAction(DeleteAccountAction.Internal.DeleteAccountComplete(result))
+            val validPasswordResult = authRepository.validatePassword(action.masterPassword)
+            if ((validPasswordResult as? ValidatePasswordResult.Success)?.isValid == false) {
```

Comment:
Shouldn't we be displaying the loading state while we validate the password?

Reply:
yeah for consistency sake. It is wicked fast since its not hitting the server. so may be a little bit of visual jank but lets give it the ol college try